### DISCUSSION TITLE
Containerd support ?

### DISCUSSION THREAD

**Discussion by aelbarkani** [okd-project/okd#2066] [General]
Hi !

I'm wondering if there is an option I can change to switch the container runtime from CRIO to ContainerD. The reason is we are using Nix inside OKD, and we would need this containerd plugin: https://github.com/pdtpartners/nix-snapshotter.

Thanks !

**Comment by GingerGeek** [comment]
I'm not aware of any work being done that would allow swapping out of container runtimes within OKD/Openshift. If we have time, this discussion may come up in triage at our next working group meeting in the new year, and I will ask others.

Only references I can find across the platform to `containerd` is for the Windows platform (not useful) and a shim present in Kata containers (sandboxed containers - you might find something useful here)

**Comment by JaimeMagiera** [comment]
Theoretically, it could be done with layering, but yes, a discussion for next year.
